src/main/java/seedu/address/ui/CommandBox.java

```diff
@@ -82,4 +108,80 @@ public interface CommandExecutor {
         CommandResult execute(String commandText) throws CommandException, ParseException;
     }
 
+    /**
+     * Contains all successfully executed commands from the last application start.
+     * Keeps track of the currently displayed command's index in the history.
+     */
+    class CommandHistory {
```

Comment:
Would CommandRetriever be a better name? So it wont be confusing with its commandHistory field